core/discovery/brokerdiscovery/repository.go

```diff
@@ -137,7 +132,7 @@ func (s *Repository) proposalPingMessage(message pingMessage) error {
 func (s *Repository) proposalWatchdog() {
 	for {
 		select {
-		case <-s.stopChan:
+		case <-s.whatchdogStop:
```

Comment:
First, a mistype. I believe that you meant `watchdog`.
Secondly, I'm not familiar with the terminology here. 

> A watchdog timer (sometimes called a computer operating properly or COP timer, or simply a watchdog) is an electronic timer that is used to detect and recover from computer malfunctions. During normal operation, the computer regularly resets the watchdog timer to prevent it from elapsing, or "timing out". If, due to a hardware fault or program error, the computer fails to reset the watchdog, the timer will elapse and generate a timeout signal. The timeout signal is used to initiate corrective action or actions. The corrective actions typically include placing the computer system in a safe state and restoring normal system operation.

https://en.wikipedia.org/wiki/Watchdog_timer

Is this the case? It appears to me that it just expires proposals and has nothing to do with system restart/recovery.

Reply:
Improved naming